src/sqlfluff/api/simple.py

```diff
@@ -123,8 +123,17 @@ def fix(
     linter = Linter(config=cfg)
 
     result = linter.lint_string_wrapped(sql, fix=True)
-    fixed_string = result.paths[0].files[0].fix_string()[0]
-    return fixed_string
+    fix_even_unparsable = cfg.get("fix_even_unparsable")
+    should_fix = fix_even_unparsable
+    if not should_fix:
+        # If fix_even_unparsable wasn't set, check for parse errors. Okay to fix
+        # if there are no parse errors.
+        _, num_filtered_parse_errors = result.check_parse_errors()
+        if not num_filtered_parse_errors:
+            should_fix = True
```

Comment:
Would this not be more readible stated as this?:

```suggestion
        if num_filtered_parse_errors == 0:
            should_fix = True
```

Reply:
Works for me!